projects/demo/src/modules/app/utils/example-content-processor/index.ts

```diff
@@ -0,0 +1,27 @@
+import {identity} from '@taiga-ui/cdk';
+import {FrontEndExample} from '../../../interfaces/front-end-example';
+import {isLess, isTS} from '../index';
+import {processLess} from './less-processor';
+import {processTs} from './typescript-processor';
+
+function getProcessor(fileName: string): (item: string) => string {
+    switch (true) {
+        case isTS(fileName):
+            return processTs;
+        case isLess(fileName):
+            return processLess;
+        default:
+            return identity;
+    }
+}
+
+export function exampleContentProcessor<T extends FrontEndExample>(content: T): T {
+    return (Object.entries(content) as [keyof T, string][]).reduce(
+        (acc, [fileName, fileContent]) => {
+            const processor = getProcessor(fileName.toString());
+
+            return {...acc, [fileName]: processor(fileContent)};
```

Comment:
```{...acc```

at each iteration you will have an object copied (shallow), from the point of view of memory usage this can lead to unpredictable consequences when you have a large object

Why not:

```ts
export function exampleContentProcessor<T extends FrontEndExample>(content: T): T {
     let processedContent: T = {};

     for (const [fileName, fileContent] of Object.entries(content)) {
       const processor = getProcessor(fileName.toString());
       processedContent[fileName] = processor(fileContent);
    }

    return processedContent;
}
```

Reply:
rewrote it as you proposed (with small changes to satisfy types)